students/TinaB/lesson05/mailroom_part3.py

```diff
@@ -0,0 +1,216 @@
+"""
+$ chmod +x mailroom_part3.py
+"""
+import datetime
+import os
+
+DONORS_DICT = {"William Gates": [326892.24, 122, 22, 12],
+               "Mark Zuckerberg": [30, 60, 65982.55],
+               "Jeff Bezos": [52636.27],
+               "Paul Allen": [877.33, 22],
+               "Steven Hawking": [326892.24, 123, 123.33, 123, 123],
+               "Justin Timberlake": [999658.25, 1233, 123]}
+
+# Menus
+
+
+def menu_selection(prompt, dispatch_dict):
+    """menu function"""
+    try:
+        while True:  # loop forever until they quit
+            response = input(prompt)
+            if dispatch_dict[response]() == "exit menu":
+                break
+    except KeyError:
+        print("Invalid input, quitting program")
+
+
+def single_print_sub_menu():
+    '''Sub Menu for Print single Thank you menu'''
+    menu_selection(SINGLE_PRINT_SUB_PROMPT, SINGLE_PRINT_SUB_DISPACTH)
+
+
+def quit_menu():
+    '''Quit menu function and method'''
+    print("Quitting this menu now")
+    return "exit menu"
+
+
+# functions for prompts ----
+
+def get_name_input():
+    ''' Function to select user input to return to print function'''
+    name_input = input(
+        "Please enter the name of the donor:  ")
+
+    for donor, donations in DONORS_DICT.items():
+        if name_input.lower() in donor.lower():
+            donor_check = input(
+                "Is this the donor you are looking for: {}?\nPlease type yes or no >> ".format(
+                    donor))
+            if donor_check == 'yes':
+                return donor
+
+    print("{} is not in our records. Let's add {} to our list!".format(
+        name_input, name_input))
+    while True:
+        new_donor_name = input(
+            "Please enter the full name of the donor or type 'exit' to quit>> ")
+        if new_donor_name == 'exit':
+            return "quit"
+        name_check = input(
+            "Does the spelling look right: {}?\nPlease type yes or no >> ".format(new_donor_name))
+        if name_check == 'yes':
+            print("Adding {} to donor list".format(new_donor_name))
+            return new_donor_name
+        elif name_check == 'exit':
+            return "quit"
+        else:  # anything other then yes
+            print(
+                "Let's try entering the donor name again. Or type 'exit' to quit to menu.")
+
+
+def send_single_thank_you():
+    """ function for sending thank you message - gets/ adds single donation and prints thank you"""
+    donor_name = get_name_input()
+    if donor_name == "quit":
+        print("No donor name entered, exiting to menu")
+    else:
+        donor_amount = check_number_input()
+
+        DONORS_DICT.setdefault(donor_name, []).append(donor_amount)
+
+        thank_you_letter = print_thank_you(donor_name, donor_amount)
+        print(thank_you_letter)
+
+
+def print_donors_names():
+    """ prints list of donors"""
+    print("\nDonors")
+    print("-"*20)
+    [print(d, "\n") for d in DONORS_DICT]
```

Comment:
comprehension is not a good pattern here since the output gets thrown out, how about this version:
```
print("\n".join(list(donors_dict)))
```